app/services/find_empty_trainees.rb

```diff
@@ -8,27 +8,21 @@ class FindEmptyTrainees
   def initialize(trainees: Trainee.all, ids_only: false)
     @trainees = trainees
     @ids_only = ids_only
-    @forms = compile_progress_validation_forms
   end
 
   def call
-    trainees.draft.filter_map do |trainee|
-      if trainee_empty?(trainee)
+    draft_trainees.filter_map do |trainee|
+      if trainee.new_and_empty?
         ids_only ? trainee.id : trainee
       end
     end
   end
 
-private
-
-  def trainee_empty?(trainee)
-    forms.all? do |progress_type, form|
-      progress_value = trainee.progress.public_send(progress_type)
-      ProgressService.call(validator: form.new(trainee), progress_value: progress_value).not_started?
-    end
-  end
-
-  def compile_progress_validation_forms
-    TrnSubmissionForm.form_validators.transform_values { |form_details| form_details[:form].constantize }
+  def draft_trainees
+    # Find all the draft trainees that do not have any degrees, disabilities and nationalities
+    trainees
+      .draft
+      .includes(:degrees, :disabilities, :nationalities)
+      .where(degrees: { id: nil }, disabilities: { id: nil }, nationalities: { id: nil })
```

Comment:
Feels like these nil checks on degrees, disabilities and nationalities should be included in the method `new_and_empty?` rather than `draft_trainees`.

If I was to use `new_and_empty?` elsewhere I think I'd assume it returned false if my trainee had a degree.

Reply:
It's a tricky one. We want to do one SQL query rather than per trainee. I think this is one of those situations where there's going to be trade offs. Maybe we could have an additional new/empty method which will also check the associated data and you'd use that one if you want a "complete" check? Currently, it only checks its own fields as the associated data can be looked up more efficiently in a batch with other trainees.

Reply:
I've moved all this logic directly into the `FindEmptyTrainees` service as it's very specific to that class's job, no need for a new method on the trainee as you're right i'd expect that to also check associated data and we have no need for it right now. Now if you want to get empty trainees you just run this service.